### PR TITLE
Bug: Remove References to Synthetic Cluster Code

### LINKED ISSUE #3240: [Dead code] Remove references to synthetic cluster code
**Please describe the Improvement and/or Feature Request**
Remove references to synthetic cluster which is no longer required or used.

https://github.com/openservicemesh/osm/search?q=getSyntheticCluster

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <aadithya794@gmail.com>

**Description**:
This PR Fixes #3240

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [X]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: pkg/envoy/cds/cluster.go [hunk 1]
```diff
@@ -75,16 +75,6 @@ func getOutboundPassthroughCluster() *xds_cluster.Cluster {
 }
 
 // getSyntheticCluster returns a static cluster with no endpoints
```
Comment (draychev) [inline pkg/envoy/cds/cluster.go:77]: @adithyaakrishna it looks like we can remove this comment as well!
```suggestion
```

**Review by draychev — APPROVED** [review]
@adithyaakrishna, thank you so much for joining us on the Open Service Mesh journey!
Really appreciate your contribution!
Love seeing removal of obsolete code!

File: pkg/envoy/cds/cluster.go (hunk 1 above)
Comment (adithyaakrishna) [reply pkg/envoy/cds/cluster.go:77]: My bad, I missed this. Will fix it ASAP

**Review by shalier — APPROVED** [review]
LGTM

**Review by shashankram — APPROVED** [review]
Hey @adithyaakrishna, this PR is good to go. I squashed your commits as per the contributing policy described in https://github.com/openservicemesh/osm/blob/main/CONTRIBUTING.md#squash-commits. In the future, make sure to squash your commits locally, thanks!

**Comment by adithyaakrishna** [comment]
Thanks @shashankram 
I had missed this squashing when I went through the Contirbuting.md file 
I'll make sure to squash them from next time onwards :)

**Comment by adithyaakrishna** [comment]
@shashankram Just to be sure, I just have to run these commands right? To squash the commits?
image

Thanks for letting me know about this, I guess I learnt something new about git 🚀 🎉

**Comment by shashankram** [comment]
> @shashankram Just to be sure, I just have to run these commands right? To squash the commits?
> image
> 
> Thanks for letting me know about this, I guess I learnt something new about git 🚀 🎉

Hey @adithyaakrishna, the above commands are required to rebase your branch and force push.

In general, this is the workflow I typically follow:
1. Make code changes locally
2. Squash local commits into a single commit before creating a PR. Refer to https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git
3. Push your local changes to your remote fork (using `git push`)
4. Get PR reviewed
5. Address comments locally
6. Commit PR feedback
7. Repeat step 2, 3, 4

**Pull request merged**